sky/spot/recovery_strategy.py

```diff
@@ -412,18 +416,17 @@ def recover(self) -> float:
             # Add region constraint to the task, to retry on the same region
             # first (if valid).
             if self._launched_cloud_region is not None:
+                assert self._launched_resources is not None
                 task = self.dag.tasks[0]
-                resources = list(task.resources)[0]
-                original_resources = resources
-
+                original_resources = task.resources
                 launched_cloud, launched_region = self._launched_cloud_region
-                new_resources = resources.copy(cloud=launched_cloud,
-                                               region=launched_region)
+                new_resources = self._launched_resources.copy(
+                    cloud=launched_cloud, region=launched_region)
```

Comment:
It seems we don't need to set the cloud and region again here, but instead, we may want to set the `zone` to None?